helm-charts/0.24.5/charts/spire/templates/spire-server.yaml

```diff
@@ -165,14 +165,19 @@ data:
 ---
 
 apiVersion: apps/v1
+{{- if eq .Values.kind "statefulset" }}
+kind: StatefulSet
+{{- else if eq .Values.kind "deployment" }}
```

Comment:
what happens if someone sets `kind: bozo`?

Do we default to StatefulSet?

Do we raise an error?

Reply:
Good point, this will be empty, which one do you like to be default? statefulset or deployment.

Reply:
BTW, empty will be statefulset, But I will make it more readable, and people will know the exact default value.